runtime/drivers/clickhouse/clickhouse.go

```diff
@@ -238,7 +252,15 @@ func (c *connection) Config() map[string]any {
 
 // Close implements drivers.Connection.
 func (c *connection) Close() error {
-	return c.db.Close()
+	return errors.Join(
+		c.db.Close(),
+		func() error {
+			if c.embed != nil {
+				return c.embed.stop()
+			}
+			return nil
+		}(),
+	)
```

Comment:
Since these calls won't be parallelized, and it's best to close the DB connection first, how about flattening the code like this?

```suggestion
	errDB := c.db.Close()

	var errEmbed error
	if c.embed != nil {
		errEmbed = c.embed.stop()
	}

	return errors.Join(errDB, errEmbed)
```